### PR TITLE
Add a list of subsystem maintainers for release

### PR DESCRIPTION
**Pull request by dasJ**
While I would be happy with a list like this, @mweinelt and @vcunat raised concerns if this would be better integrated with the teams list.
I agree with them for most of this list, but there are also parts of the list where I think this model would not hold very well, especially the first section.

This issue is a draft and hopefully we can start a discussion and come to a solution people like.
I'm not against merging the list into the teams list but we need to find a way to structure it properly and a script to render a textual version of the list for posting on Discourse.

cc @jonringer @tomberek @Mic92, maybe you have some ideas

### CODE REVIEW

**Comment by mweinelt** [comment]
As stated I'm a for reusing the teams infrastructure that we have for maintainers and maybe a flag that declares whether it should be pinged for this purpose?

Also I'm reiterating that I think I'm one of the primary python package set maintainers these days, but I'm very reluctant to adding myself as a CODEOWNER for `pkgs/development/python-modules`, because I wouldn't be able to deal with the influx of notifications, when the directory sees 500-700 PRs per month.

**Comment by dasJ** [comment]
I redid it with the maintainers list. Is this what you were looking for?

File: maintainers/team-list.nix
Comment (Mic92) [inline maintainers/team-list.nix:76]:
```suggestion
      matthewbauer
      mic92
```

File: maintainers/team-list.nix
Comment (Mic92) [inline maintainers/team-list.nix:490]: Not sure where the Release engineer came from.
```suggestion
      mic92 # Release editor
```

File: maintainers/team-list.nix
Comment (dasJ) [reply maintainers/team-list.nix:545]: I don't know about andir but I know that I don't need to mention flokli because he's already in the GitHub Team ;)

File: maintainers/team-list.nix
Comment (dasJ) [reply maintainers/team-list.nix:490]: From my memory probably… I removed all users in this team and switched to the GitHub team so we don't need to update this for every release (not pushed yet) ;)

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: maintainers/team-list.nix
Comment (mweinelt) [reply maintainers/team-list.nix:545]: I think stepping away from the latest systemd update meant exactly that.

File: maintainers/scripts/feature-freeze-teams.pl
Comment (flokli) [inline maintainers/scripts/feature-freeze-teams.pl:1]: Ugh, perl. Can we write this in something more approachable, like python, so people feel empowered to change this script?

I think this was one of the main reasons on why we switched to using python for the VM test framework.

Comment (dasJ) [reply maintainers/scripts/feature-freeze-teams.pl:1]: Feel free

File: maintainers/team-list.nix
Comment (dasJ) [reply maintainers/team-list.nix:49]: Hm, but setting the bool requires setting a team name as well. I don't really see a point in having both since you will always have to set the string and also another bool that is always `true` when the name is set.

File: maintainers/team-list.nix
Comment (dasJ) [reply maintainers/team-list.nix:70]: I wanted to take the fear of people that they will now get randomly added as the maintainers of packages because it's part of their subsystem.
Changing the scope later is never a problem and can freely be done if people want to use the team for other purposes.

File: maintainers/team-list.nix
Comment (dasJ) [reply maintainers/team-list.nix:197]: Why not? Having a list gives us more flexibility for future changes.

File: maintainers/team-list.nix
Comment (mweinelt) [reply maintainers/team-list.nix:70]: Setting someone else as maintainer without their approval is a big no-no anyway, so I don't see the problem.

File: maintainers/team-list.nix
Comment (mweinelt) [reply maintainers/team-list.nix:49]: But now every other script that follows will have to use the `featureFreezeTeamName` if they want a pretty name. Or they have to refactor the structure another time.

File: maintainers/team-list.nix
Comment (dasJ) [reply maintainers/team-list.nix:545]: I will remove him

File: maintainers/team-list.nix
Comment (dasJ) [reply maintainers/team-list.nix]: Is this better? I also changed `scope` to `description` because it's a pattern I've seen more often in nixpkgs than `scope`

Comment (mweinelt) [reply maintainers/team-list.nix]: I would've just made the human-readable name mandatory. Setting it is a small ask for everyone touching this file.

**Comment by dasJ** [comment]
Okay, I hope this is what you were after.
Generated list looks like this (let's hope the code block prevents pings):
```
$ scripts/feature-freeze-teams.pl
GitHub username: dasJ
GitHub personal access token (no echo):

PostgreSQL: @thoughtpolice
Cleanup: @ajs124
C: @matthewbauer @mic92
Docs: @NixOS/nixos-release-managers
PHP: @NixOS/php @ma27
Erlang: @NixOS/beam
Python: @fridh @mweinelt @jonringer
Podman: @NixOS/podman
Coq: @CohenCyril @Zimmi48 @siraben @vbgl
Darwin: @NixOS/darwin-maintainers
Go: @c00w @cstrahan @Frostman @kalbasit @mic92 @orivej @rvolosatovs @zowoq
DockerTools: @roberth @utdemir
NixOS Modules / internals: @ericson2314 @infinisil @alyssais @roberth
Rust: @andir @lnl7 @mic92 @zowoq
Marketing: @garbas @tomberek
Blockchains: @mmahut @RaghavSood
Emacs: @adisbladis
Haskell: @NixOS/haskell
Everyone else: @NixOS/nixpkgs-committers @NixOS/release-engineers
NixOS tests: @tfc
Docs: @ryantm
gnome: @NixOS/gnome @dasj19 @maxeaubrey
Perl: @stigtsp
systemd: @NixOS/systemd
Mobile: @samueldr
Pantheon: @NixOS/pantheon
Vim/Neovim: @jonringer @softinio @teto
Ruby: @marsam
Dhall: @Gabriel439 @ehmry
Cinnamon: @mkg20001
R: @bcdarwin @jbedo
Nix/nix-cli ecosystem: @Profpatsch @edolstra @grahamc @nbp
Qt / KDE: @NixOS/qt-kde
Bazel: @mboes @marsam @uri-canva @avdv @olebedev @groodt @aherrmann @ylecornec
```

File: maintainers/team-list.nix
Comment (jtojnar) [inline maintainers/team-list.nix:265]: Should be written in all-caps.

File: maintainers/team-list.nix
Comment (jtojnar) [inline maintainers/team-list.nix:370]: Also all-caps.

File: maintainers/team-list.nix
Comment (jtojnar) [inline maintainers/team-list.nix:510]: I do not see why this should not be used for package maintainers if Thomas wanted to do so. I would suspect most people who have not yet switched to teams for their package collections are just too busy to carry that out.

**Review by mweinelt — APPROVED** [review]

File: maintainers/team-list.nix
Comment (mweinelt) [reply maintainers/team-list.nix:510]: I had similar concerns about the wording. https://github.com/NixOS/nixpkgs/pull/165978#discussion_r836402127

File: maintainers/team-list.nix
Comment (jtojnar) [inline maintainers/team-list.nix:639]: Xfce is most commonly spelt in title-case

File: maintainers/team-list.nix
Comment (jtojnar) [inline maintainers/team-list.nix:582]: missing h

File: maintainers/team-list.nix
Comment (dasJ) [reply maintainers/team-list.nix:510]: Do you have an alternative wording? Because this is the best that springs out of my tired brain

**Comment by FRidh** [comment]
Partially related issue https://github.com/NixOS/nixos-homepage/issues/645

**Comment by dasJ** [comment]
Since I want to have this merged until the weekend, I wrote scopes for all teams.
Is everyone happy with these?

**Pull request merged**
